### PR TITLE
Refine the vineyard client wrapper to handle the corner case and avoid the block of IO

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

Related issue number
--------------------

Fixes #issue number

### CODE REVIEW

File: python/vineyard/core/client.py [hunk 1]
```diff
@@ -675,7 +687,10 @@ def put(
         try:
             return put(self, value, builder, persist, name, **kwargs)
         except NotEnoughMemoryException as exec:
+            print("called _find_the_most_available_memory_instance...")
             instance_id = self._find_the_most_available_memory_instance()
+            print("after called _find_the_most_available_memory_instance...")
+            previous_compression_state = self.compression
```
Comment (sighingnow) [inline python/vineyard/core/client.py:693]: Delete these two debug logs.

File: python/vineyard/core/client.py [hunk 2]
```diff
@@ -622,27 +635,33 @@ def _find_the_most_available_memory_instance(self) -> int:
             int: The instance id with the most available memory.
             if only have one instance, return -1
         """
+        os.environ['VINEYARD_RPC_SKIP_RETRY'] = '1'
```
Comment (sighingnow) [inline python/vineyard/core/client.py:638]: Use `with vineyard.envvars()....` to avoid the global variable been populated.

Comment (dashanji) [reply python/vineyard/core/client.py]: Make sense. Thanks.

File: python/vineyard/core/client.py (hunk 1 above)
Comment (dashanji) [reply python/vineyard/core/client.py]: Ok.

**Pull request merged**